README.md

```diff
@@ -2,15 +2,211 @@
+kube-node-lease   Active   99d
+kube-public       Active   99d
+kube-system       Active   99d
+secrets-system    Active   13m
+````
+
+Show Secrets
+```shell
+kubectl -n secrets-system get all
+```
+Output:
+```
+NAME                                                          DESIRED   CURRENT   READY   AGE
+replicaset.apps/external-secrets-78986ddfdf                   1         1         1       14m
+replicaset.apps/external-secrets-cert-controller-79688fbc89   1         1         1       14m
+replicaset.apps/external-secrets-webhook-887d856fb            1         1         1       14m
+```
+### 6. Issue
+An Output of issue can be found at `Output.md`
+[Execution log](Output.md)
+
+There is also Warning from "externalsecret-validate" with message "ca cert not yet ready" from object `
+validating-webhook-configuration`
+
+### 7. Stop and clean up all
+#### 7.a kill vault server
+```shell
+pkill vault;
+ps aux | grep vault | grep -v grep
+```
+
+#### 7.b clean up k8s
+##### 7.b.1. remove the finalizer from "argocd"
+```shell
+kubectl get namespace argocd -o json >tmp.json;
+```
+
+##### 7.b.2. Install jq
 ```shell
-$ vault server -dev -dev-root-token-id="education"
+brew install jq
 ```
-3. Export your new Vault address to the shell
-4. Initialize your infrastructure
+
+##### 7.b.3. Replace the finalizer values 
 ```shell
-$ terraform init -upgrade
-$ terraform apply
+(jq '.spec.finalizers = []' < tmp.json) > new_tmp.json
 ```
-5. Check your infrastructure (`k9s`, `kubectl`, Lens...)
+Reference:
+* redirect file to jq: https://stackoverflow.com/questions/53280090/parse-error-invalid-numeric-literal-at-line-2-column-0/53280264#53280264
+* sed is only for text replacement https://www.cyberciti.biz/faq/how-to-use-sed-to-find-and-replace-text-in-files-in-linux-unix-shell/
+* remove the namespace with finalizers: https://phoenixnap.com/kb/kubernetes-delete-namespace
+
+##### 7.b.4. patch with forwarding
+1. Open the proxy from a separate terminal
+```shell
+kubectl proxy
+```
+
+2. Call the curl api to remove the finalizers on the argocd namespace
+```
+curl -k -H "Content-Type: application/json" -X PUT --data-binary @new_tmp.json http://127.0.0.1:8001/api/v1/namespaces/argocd/finalize
+```
+
+The pattern used is:
+```
+curl -k -H "Content-Type: application/json" -X PUT --data-binary @tmp.json http://127.0.0.1:8001/api/v1/namespaces/[your-namespace]/finalize
+````
+
+<!-- this patch method doesn't work
+2. Patch namespace
+```shell
+kubectl patch namespace argocd --patch '{"spec": {"finalizers":[]}}' 
+```
+if patch don't work
+
+```
+kubectl edit namespace argocd
+```
+replace
+```
+spec:
+  finalizers:
+  - kubernetes
+with
+spec:
+  finalizers:  
+```
+-->
+
+3. After change the finalizers of argocd namespace, remove all the namespace
+```shell
+kubectl delete namespace apps-system argocd secrets-system
+```
+
+4. terminate the kubectl proxy with `ctl + c`
+5. remove the tmp.json files
+```shell
+rm tmp.json;
+rm new_tmp.json
+```
```

Comment:
The idea of the exercise is to control **everything** via GitOps.

Patching manually a resource goes against the purpose of it.